File: dsa/week(2)/Studentclass.py
class Student:
    def __init__(self):
        self.roll_number = None
        self.__age = None  
    def display(self):
        print(f"{self.__age} {self.roll_number}")
    def get_age(self):
        return self.__age
    def set_age(self, age, password):
        if password != 123:
            return
        if age < 0:
            return
        self.__age = age
def main():
    s1 = Student()
    s2 = Student()
    s3, s4, s5 = Student(), Student(), Student()
    # s1.age = 24  # This will raise an AttributeError because age is private
    s1.roll_number = 101
    print(f"S1 age: {s1.get_age()}")
    print(f"S1 Roll number: {s1.roll_number}")
    s1.display()
    s2.display()
    print(s1.roll_number)
    # Create objects dynamically
    s6 = Student()
    # s6.age = 23  # This will raise an AttributeError because age is private
    s6.roll_number = 105654
    s6.display()
    print(f"S6 age: {s6.get_age()}")
    # s6.age = 23  # This will raise an AttributeError because age is private
    s6.roll_number = 104
    s6.display()
    s1 = Student()
    s2 = Student()
    s1.set_age(20, 123)
    s2.set_age(24, 123)
    s1.display()
    s2.display()


if __name__ == "__main__":
    main()

File: dsa/week(2)/Studentclass.js
class Student {
  constructor() {
    this.rollNumber = null;
    this._age = null;
    this.display = function() {
      console.log(this._age, this.rollNumber);
    };
    this.getAge = function() {
      return this._age;
    };
    this.setAge = function(a, password) {
      if (password !== 123) {
        return;
      }
      if (a < 0) {
        return;
      }
      this._age = a;
    };
  }
}

// Main function
function main() {
  let s1 = new Student();
  let s2 = new Student();
  let s3 = new Student();
  let s4 = new Student();
  let s5 = new Student();

 
  s1.rollNumber = 101;

  console.log("S1 age : ", s1.getAge());
  console.log("S1 Roll number : ", s1.rollNumber);
  s1.display();
  s2.display();

 
  let s6 = new Student();
  s6.rollNumber = 104233;

  // Display age and rollNumber for s6
  s6.display();
  console.log("S6 age : ", s6.getAge());

  // Set rollNumber for s6


  // Create new objects
  let s7 = new Student();
  let s8 = new Student();

  // Set age for s7 and s8
  s7.setAge(20, 123);
  s8.setAge(24, 123);

  // Display age and rollNumber for s7 and s8
  s7.display();
  s8.display();
}

// Call the main function
main();